comunicatii/src/main/java/code4romania/czl/scrapers/comunicatii/ScraperComunicatii.java

```diff
@@ -26,216 +26,282 @@
-    }
+	}
+
+	private static void extractDate(ComunicatiiBean cBean) {
+		Pattern p = Pattern.compile(DATE_REGEX);
+
+		Matcher matcher = p.matcher(cBean.getDescription());
+		while (matcher.find()) {
+			String date = cBean.getDate();
+			cBean.setDate(matcher.group(), "dd.MM.yyyy");
+			System.out.println(matcher.group());
+		}
+	}
+
+	private static String extractFeedbackDate(String description,
+			String dateFeedbackRegex) {
+		Pattern pFeedback = Pattern.compile(dateFeedbackRegex);
+		Pattern pDate = Pattern.compile(DATE_REGEX);
+		Matcher m = pFeedback.matcher(description);
+		try {
+			m.find();
+			Matcher mDate = pDate.matcher(m.group());
+			mDate.find();
+			return mDate.group();
+		} catch (IllegalStateException e) {
+			e.printStackTrace();
+		}
+		return "";
+	}
+
+	private static String getTextFromNode(Node cNode) {
+		StringBuffer text = new StringBuffer();
+		cNode.childNodes().forEach(node -> {
+			switch (node.nodeName()) {
+			case "#text":
+				text.append(node.toString());
+				break;
+			case "a":
+				text.append(getTextFromNode(node));
+				break;
+			}
+		});
+		return text.toString();
+	}
+
+	private static void selectLinks(Document document) {
+		Elements links = document.select("a");
+		List<ComunicatiiBean> outputList = new ArrayList<ComunicatiiBean>();
+		for (Element link : links) {
+			ComunicatiiBean cBean = new ComunicatiiBean();
+			outputList.add(cBean);
+			ComunicatiiBean.Document cDoc = new ComunicatiiBean.Document();
+			extractDocument(link, cBean, cDoc);
+
+			link.textNodes().forEach(
+					linknode -> System.out.println("link node: "
+							+ linknode.toString()));
+
+			System.out.println(cBean);
+		}
+	}
+
+	private boolean isSeparator(Element e) {
+		e.text();
+		return false;
+	}
+
+	private void processRelatedElements(List<Element> elements) {
+	}
+
+	private static void extractDocument(Element link, ComunicatiiBean cBean,
+			ComunicatiiBean.Document cDoc) {
+		String docUrl = link.attr("href");
+		cDoc.setUrl(docUrl);
+		try {
+			cDoc.setType(docUrl.substring(docUrl.lastIndexOf('.')));
+
+		} catch (Exception e) {
+			e.printStackTrace();
+		}
+		cBean.getDocuments().add(cDoc);
+	}
+
+	public static String calculateFeedbackDays(String dataPublicare,
+			String dataFeedback) {
+		// Converting date to Java8 Local date
+		DateTimeFormatter fmt = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+		try {
+			LocalDate startDate = LocalDate.parse(dataPublicare, fmt);
+			LocalDate endDate = LocalDate.parse(dataFeedback, fmt);
+			// Range = End date - Start date
+			Long range = ChronoUnit.DAYS.between(startDate, endDate);
+
+			return range.toString();
+		} catch (Exception e) {
+			e.printStackTrace();
+		}
+		return "0";
+	}
+
+	public static String deductType(String description) {
+		String type = "";
+		if (description.contains("Proiect de Lege")
+				|| description.contains("Proiect de lege")) {
+			type = type + "LEGE";
+		}
+		if (description.contains("Ordonanță")
```

Comment:
Doua sugestii:

* transforma descrierea in lowercase
* inlocuieste diacriticele

Cautarea ar fi atunci mai simpla.